lib/doorkeeper_patches/models/application.rb

```diff
@@ -11,16 +11,22 @@ class Doorkeeper::Application
                                  allow_blank: true,
                                  message: 'Logo url must begin with https'
 
-  validates_acceptance_of :terms_of_service_accepted, if: :federal_agency,
-                                                      accept: true,
-                                                      allow_nil: false,
-                                                      message: :federal_agency_tos_required,
-                                                      tos_link: ->(values) { Rails.application.routes.url_helpers.legal_path(anchor: 'terms-of-service') }
-
-  validates_presence_of :organization, if: :federal_agency,
-                                       accept: true,
-                                       allow_nil: false,
-                                       message: :federal_agency_org_required
+  validates_acceptance_of :terms_of_service_accepted,
+    if: :federal_agency, accept: true, allow_nil: false,
+    message: :federal_agency_tos_required,
+    tos_link: ->(values) { Rails.application.routes.url_helpers.legal_path(anchor: 'terms-of-service') }
```

Comment:
if you look in `en.yml` for `federal_agency_tos_required`, you'll notice it has a `tos_link` attribute. the lambda in the diff above will call `legal_path(anchor: 'terms-of-service')` get evaluated at run time to `/legal#terms-of-service`, and then the active record error message code will call `I18n.t(:message, tos_link: ...)`